File: leetcode-submissions/2461_Maximum_Sum_of_Distinct_Subarrays_With_Length_K.ts
function maximumSubarraySum(nums: number[], k: number): number {
    let i:number =0;
    let j:number =0;
    let result:number =0;

    const set=new Set<number>();
    let currentWindowSum:number =0;

    while(j<nums.length){
        while(set.has(nums[j])){
            set.delete(nums[i]);
            currentWindowSum-=nums[i];
            i++;
        }
        currentWindowSum+=nums[j];
        set.add(nums[j]);
        if(j-1+1===k){
            result=Math.max(result,currentWindowSum);
            currentWindowSum-=nums[i];
            set.delete(nums[i]);
            i++;
        }
        j++;

    }
    return result ;
};

File: leetcode-submissions/2461_Maximum_Sum_of_Distinct_Subarrays_With_Length_K.cpp
class Solution {
public:
    long long maximumSubarraySum(vector<int>& nums, int k) {
        int i=0,j=0,currentWindowSum=0,result=0;
        set<int>s;
        int n=nums.size();
        while(j<n){
            while(s.count(nums[j])){
                currentWindowSum-=nums[i];
                s.erase(nums[i]);
                i++;

            }
            s.insert(nums[j]);
            currentWindowSum+=nums[j];
            if(j-i+1==k){
                result=max(result ,currentWindowSum);
                currentWindowSum-=nums[i];
                s.erase(nums[i]);
                i++;
            }
            j++;
        }
return result;
    }
};